### PR TITLE
Refactor package partial links (_side_links)

### PR DESCRIPTION
**Pull request by vpereira**
Refactor the partial _side_links for packages, breaking it in smaller partials

### CODE REVIEW

**Comment by vpereira** [comment]
[P5: Refactor Package partials links](https://trello.com/c/XV8by87E/853-refactor-package-partials-links)

File: src/api/app/views/webui2/webui/package/_side_links.html.haml [hunk 1]
```diff
@@ -1,66 +1,24 @@
 -# FIXME: refactor this file
 %ul.list-unstyled
   - if failures > 0
-    %li
-      %i.fas.fa-times-circle.text-danger{ title: 'Errors' }
-      = failures
-      = link_to "error#{failures == 1 ? '' : 's'}", project_monitor_path(project: project, pkgname: package.name, succeeded: 0,
-                                                                         blocked: 0, finished: 0, signing: 0, dispatching: 0,
-                                                                         scheduled: 0, building: 0)
+    = render partial: 'show_failures', locals: { failures: failures, pkgname: package.name, project: project }
```
Comment (DavidKang) [inline src/api/app/views/webui2/webui/package/_side_links.html.haml]: [Nitpick]: Could you change `pkgname` to `package_name`, please

**Comment by dmarcoux** [comment]
I would squash the commits, we don't need this amount of details when it's about splitting a partial. How about this `Split partial package/_side_links.html.haml into smaller partials`

**Comment by vpereira** [comment]
> I would squash the commits, we don't need this amount of details when it's about splitting a partial. How about this `Split partial package/_side_links.html.haml into smaller partials`

first I have to figure out why the tests aren't green. Therefore I prefer to have it now as single commits. Before merge I will squash it. thank you

File: src/api/app/views/webui2/webui/package/_side_links.html.haml (hunk 1 above)
Comment (vpereira) [reply src/api/app/views/webui2/webui/package/_side_links.html.haml]: if you are saying to change it in the view, doable nitpick, but I won't touch in the controller https://github.com/vpereira/open-build-service/blob/da08d94faa5c84c1ba3bc175c38c61584976cad2/src/api/app/controllers/webui/project_controller.rb#L390-L391 where the GET param name is ```pkgname```

Comment (DavidKang) [reply src/api/app/views/webui2/webui/package/_side_links.html.haml]: @vpereira, only in the view :+1:

File: src/api/app/views/webui2/webui/package/_show_failures.html.haml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+%li
+  %i.fas.fa-times-circle.text-danger{ title: 'Errors' }
+  = failures
+  = link_to "error#{failures == 1 ? '' : 's'}", project_monitor_path(project: project, pkgname: package_name, succeeded: 0,
```
Comment (dmarcoux) [inline src/api/app/views/webui2/webui/package/_show_failures.html.haml]: Nice use case for [pluralize](https://api.rubyonrails.org/classes/ActionView/Helpers/TextHelper.html#method-i-pluralize)

Comment (DavidKang) [reply src/api/app/views/webui2/webui/package/_show_failures.html.haml]: @dmarcoux, good catch

**Pull request merged**